solarforecastarbiter/metrics/tests/test_calculator.py

```diff
@@ -371,8 +748,75 @@ def test_apply_deterministic_metric_func(metric, fx, obs, ref_fx, norm, expect,
 def test_apply_deterministic_bad_metric_func():
     with pytest.raises(KeyError):
         calculator._apply_deterministic_metric_func('BAD METRIC',
-                                                    pd.Series(dtype=float),
-                                                    pd.Series(dtype=float))
+                                                    np.array([]),
+                                                    np.array([]))
+
+
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+@pytest.mark.parametrize('metric,fx,fx_prob,obs,ref_fx,ref_fx_prob,expect', [
+    ('bs', [], [], [], None, None, np.NaN),
+    ('bs', [1, 1, 1], [100, 100, 100], [1, 1, 1], None, None, 0.),
+
+    # Briar Skill Score with no reference
+    ('bss', [1, 1, 1], [100, 100, 100], [0, 0, 0],
+        None, None, 1.),
+    ('bss', [1, 1, 1], [100, 100, 100], [1, 1, 1],
+        [0, 0, 0], [100, 100, 100], 1.),
+
+    ('rel', [1, 1, 1], [100, 100, 100], [1, 1, 1], None, None, 0.),
+    ('res', [1, 1, 1], [100, 100, 100], [1, 1, 1], None, None, 0.),
+    ('unc', [1, 1, 1], [100, 100, 100], [1, 1, 1], None, None, 0.),
+
+    # CRPS single forecast
+    ('crps', [[1]], [[100]], [[0]], None, None, 0.),
+    # CRPS mulitple forecasts
+    ('crps', [[1, 1, 1], [2, 2, 2], [3, 3, 3]],
+             [[100, 100, 100], [100, 100, 100], [100, 100, 100]],
+             [0, 0, 0], None, None, 0.)
+])
+def test_apply_probabilistic_metric_func(metric, fx, fx_prob, obs,
+                                         ref_fx, ref_fx_prob, expect,
+                                         create_dt_index):
+    if metric == 'crps':
```

Comment:
does crps need to be handled in a special way? I don't see anything like this in the calculator code but maybe I'm missing something?

Reply:
That's an artifact. I removed it